src/Infrastructure/Service/TaskManager.php

```diff
@@ -55,6 +64,10 @@ public function initialize(array $tasks): array
                 continue;
             }
 
+            if ($task instanceof TaskSetInterface) {
```

Comment:
it's not correct way to extract taskset. you did it from the persistence repository. This condition will be false every time. you need to move it to the task saver. When we save tasks from yaml repository.